
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Utility &#8212; DEep-learning and SIngle-cell based DEconvolution</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.acff12b8f9c144ce68a297486a2fa670.css" type="text/css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <link rel="shortcut icon" href="../_static/logo.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Change Log" href="../changelog.html" />
    <link rel="prev" title="Plot" href="plot.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      <img src="../_static/logo.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">DEep-learning and SIngle-cell based DEconvolution</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <div class="bd-toc-item active">
        <ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../installation.html">
   Installation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../usage.html">
   Usage
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../datasets.html">
   Datasets
  </a>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="../functions.html">
   Functions
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="single_cell.html">
     Single Cell Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="bulk_cell.html">
     Bulk Cell Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="simulation.html">
     Simulation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="deconvolution.html">
     Deconvolution
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="plot.html">
     Plot
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     Utility
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../changelog.html">
   Change Log
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../contact.html">
   Contact
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://github.com/OnlyBelter/DeSide">
   GitHub repository
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/func/utility.md.txt"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/OnlyBelter/DeSide"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/OnlyBelter/DeSide/issues/new?title=Issue%20on%20page%20%2Ffunc/utility.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/OnlyBelter/DeSide/edit/master/docs/func/utility.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="module-deside.utility">
<span id="utility"></span><h1>Utility<a class="headerlink" href="#module-deside.utility" title="Permalink to this headline">¶</a></h1>
<dl class="py function">
<dt id="deside.utility.log_exp2cpm">
<code class="sig-prename descclassname"><span class="pre">deside.utility.</span></code><code class="sig-name descname"><span class="pre">log_exp2cpm</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exp_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log_base</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">correct</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span> &#x2192; <span class="pre">pandas.core.frame.DataFrame</span><a class="headerlink" href="#deside.utility.log_exp2cpm" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert log2(CPM + 1) to non-log space values (CPM / TPM)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>exp_df</strong> – samples by genes</p></li>
<li><p><strong>log_base</strong> – the base of log transform</p></li>
<li><p><strong>correct</strong> – plus 1 for avoiding log transform 0</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>counts per million (CPM) or transcript per million (TPM)</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="deside.utility.non_log2cpm">
<code class="sig-prename descclassname"><span class="pre">deside.utility.</span></code><code class="sig-name descname"><span class="pre">non_log2cpm</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">exp_df</span></span></em><span class="sig-paren">)</span> &#x2192; <span class="pre">pandas.core.frame.DataFrame</span><a class="headerlink" href="#deside.utility.non_log2cpm" title="Permalink to this definition">¶</a></dt>
<dd><p>Normalize gene expression to CPM / TPM for non-log space</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>exp_df</strong> – gene expression profile in non-log space, sample by gene</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>counts per million (CPM) or transcript per million (TPM)</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="deside.utility.non_log2log_cpm">
<code class="sig-prename descclassname"><span class="pre">deside.utility.</span></code><code class="sig-name descname"><span class="pre">non_log2log_cpm</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file_path</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">result_file_path</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">transpose</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">correct</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#deside.utility.non_log2log_cpm" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert non-log expression data to log2(CPM + 1) or log2(TPM + 1)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_file_path</strong> – non-log space expression file</p></li>
<li><p><strong>result_file_path</strong> – file path, samples by genes</p></li>
<li><p><strong>transpose</strong> – if input file is samples by genes, set to False, otherwise set to True</p></li>
<li><p><strong>correct</strong> – plus 1 for avoiding log transform 0</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>log2(CPM + 1) or save result to file</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt id="deside.utility.read_data_from_h5ad">
<code class="sig-prename descclassname"><span class="pre">deside.utility.</span></code><code class="sig-name descname"><span class="pre">read_data_from_h5ad</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">h5ad_file_path</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> &#x2192; <span class="pre">dict</span><a class="headerlink" href="#deside.utility.read_data_from_h5ad" title="Permalink to this definition">¶</a></dt>
<dd><p>Read simulated bulk gene expression profiles (GEPs) from .h5ad file</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>h5ad_file_path</strong> – the file path of simulated bulk GEPs file (.h5ad)</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>a dict contains bulk GEPs (bulk_exp) and cell fractions used when generate this dataset (cell_frac)</p>
</dd>
</dl>
</dd></dl>

</div>


              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="plot.html" title="previous page">Plot</a>
    <a class='right-next' id="next-link" href="../changelog.html" title="next page">Change Log</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Xin Xiong<br/>
        
            &copy; Copyright 2021, iSynBio.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>